src/pyOpenMS/pyopenms/dataframes.py

```diff
@@ -304,6 +304,61 @@ def get_df(self, long : bool = False):
 
         return pd.DataFrame(data=((spec.getRT(), *spec.get_peaks()) for spec in self), columns=cols)
 
+    def get_massql_df(self):
+        """Exports data from MSExperiment to pandas DataFrames to be used with MassQL.
+
+        The Python module massql allows queries in mass spectrometry data (MS1 and MS2
+        data frames) in a SQL like fashion (https://github.com/mwang87/MassQueryLanguage).
+        
+        Returns:
+        ms1_df (pandas.DataFrame): peak data of MS1 spectra
+        ms2_df (pandas.DataFrame): peak data of MS2 spectra with precursor information
+        """
+        self.updateRanges()
+
+        def get_polarity(spectrum):
+            polarity = spectrum.getInstrumentSettings().getPolarity()
+            if polarity == IonSource.Polarity.POLNULL:
+                return 0
+            elif polarity == IonSource.Polarity.POSITIVE:
+                return 1
+            elif polarity == IonSource.Polarity.NEGATIVE:
+                return 2
+
+        def get_peak_arrays_from_spec(spec, scan_num):
+            if spec.getMSLevel() == 2:
+                ms2_data = (spec.getPrecursors()[0].getMZ(), self.getPrecursorSpectrum(scan_num)+1, spec.getPrecursors()[0].getCharge())
+            else:
+                ms2_data = ()
+            for i, peak in enumerate(spec):
+                peak_data = (peak.getIntensity(), peak.getMZ(), scan_num+1, spec.getRT()/60, get_polarity(spec))
```

Comment:
```suggestion
                peak_data = (peak.getIntensity(), peak.getMZ(), scan_num+1, spec.getRT()/60.0, get_polarity(spec))
```

Reply:
I think we have a numpy function getPeakData... could this be used to make this faster?

Reply:
Yes, extract the full numpy arrays for mz and inty, generate equally sized numpy arrays for RT/60, scannum+1 and polarity (e.g. `a=np.empty(len(mzvals)); a.fill(RT/60)`), concatenate rowwise and then transpose.
Should be fastest, since all is in numpy

Reply:
Maybe add two more np.empty rows for i_norm and tic_i_norm. So you just have to fill it later and don't need to use insert.

Reply:
Intersting, this would create the ndarray for the spectrum without numpy.asarray from a list of python lists. Will try...